### ISSUE TITLE
Problems running on mobile

### ISSUE THREAD

**Issue by electr0sheep** [meteor/todos#107]
The cordova-build directory must be removed `rm -r .meteor/local/cordova-build` before each mobile run `meteor run android` `meteor run android-device` `meteor run ios`

`meteor run android` does not start app on emulator

When executing on a device `meteor run android-device`, the app has trouble connecting to the database.

When executing `meteor run ios`, the app hangs on the meteor splash screen.

**Comment by tmeasday** [comment]
@martijnwalraven could you take a look at this?

**Comment by martijnwalraven** [comment]
@electr0sheep: Why do you have to remove the cordova-build directory before each mobile run? What happens if you don't do this?

The only thing I've been able to reproduce is that it takes a long time for the launch screen to hide.

It seems [this code](It seems https://github.com/meteor/meteor/blob/devel/packages/launch-screen/default-behavior.js#L22) in the `launch-screen` package is never called, so we rely on the 6 second timeout below.

@tmeasday: Do you have any idea why the `Template.body.onRendered` callback would not be invoked with the todos app?

**Comment by tmeasday** [comment]
@martijnwalraven maybe because there's no `<body>` tag? Where do we use `Template.body.onRendered`? Is that part of the launch screen package?

**Comment by martijnwalraven** [comment]
@tmeasday: Sorry, I made a type in the markdown of my response. There should be a link to the code in the `launch-screen` package. 

It seems this behavior may be because of the use of `blaze-layout`? https://github.com/kadirahq/blaze-layout/issues/27

**Comment by tmeasday** [comment]
Yeah. Probably the solution is to just whack down an empty `<body>` in `client/main.html`

**Comment by electr0sheep** [comment]
@martijnwalraven If I do not remove the build directory I get the following error when building for mobile (android and ios):
```bash
shell.js: internal errorject
Error: ENOENT, no such file or directory '/Users/(my name)/todos/.meteor/local/cordova-build/www/*'
```

Some additional info: I am on OS X El Capitan version 10.11 Beta.
Furthermore, I experienced no issues when I ran the simple-todos tutorial.

Also, as far as the trouble connecting to the database, I experienced no issues with the tutorial, and am connected to the same network which does not have client isolation, etc.

**Comment by martijnwalraven** [comment]
@electr0sheep: Hmm, this is really surprising. `shelljs` is used by Cordova (not our code), and this seems like it is trying to copy the assets, probably to the platform-specific directories (like `platforms/ios/www`).

I've never seen this error, so I suspect it may have something to do with your setup. Maybe a permissions problem of some kind. Have you tried to clone `todos` to a new directory?

As for the connection issue, do you see any errors either in the native logs or through remote debugging? (See the [mobile guide](http://guide.meteor.com/mobile.html#logging-and-remote-debugging) for instructions.)

**Comment by electr0sheep** [comment]
@martijnwalraven I don't think it would be permissions. I put the tutorial project in the same directory e.g. ```/Users/michaeldegraw/```
However...I just realized something else I forgot to mention before. Before it would even run, I had to ```npm install```. If I attempt to run the project before executing that command, it would give me the following errors:
```shell
Unable to resolve some modules:
"faker" in /Users/michaeldegraw/Desktop/todos/imports/api/todos/todos.js (web.browser)
Consider running: meteor rmnpm install --save faker

Unable to resolve some modules:
"faker" in /Users/michaeldegraw/Desktop/todos/imports/api/todos/todos.js(web.cordova)
Consider running: meteor npm install --save faker

Unable to resolve some modules:
"faker" in /Users/michaeldegraw/Desktop/todos/imports/api/todos/todos.js(os.osx.x86_64)
Consider running: meteor npm install --save faker
```

Now, if I run meteor npm install --save faker, it gives me the following warning:
```bash
npm WARN package.json @ No repository field.
```
 and then gives me the following error when I attempt to run afterwards:

```bash
=>Errors prevented startup:

While minifying app stylesheets:
error: Can't load npm module 'autoprefixer' from
packages/minifier-postcss_plugin.js. Check your Npm.depends().
```

I can't believe I forgot to mention that before.

**Comment by tmeasday** [comment]
@martijnwalraven: I'm seeing this also:

```
shell.js: internal errorject                 \
Error: ENOENT, no such file or directory '/Users/tom/Meteor/todos/.meteor/local/cordova-build/www/*'
```

**Comment by martijnwalraven** [comment]
@tmeasday: I now also saw this after the first run (I think). It turns out it is a bug in `shelljs` 0.6.0 (https://github.com/shelljs/shelljs/issues/389). The dev bundle contains `shelljs` 0.3.0 (which is what `cordova-lib` specifies), but `node_modules` in the app directory contains 0.6.0 and apparently is the one that gets loaded. 

Not sure what to do about this, it seems like a general issue (local Node packages overriding the ones in the dev bundle).

**Comment by tmeasday** [comment]
Whoa... say what!? That seems really scary

**Comment by timmakken** [comment]
I sort of have the same problem, i forgot to `npm install ` first. then got errors after doing it thus deleted node_modules folder and did `meteor npm install` to make a fresh clone work.

The app now works fine in the browser but now it does not seem to run on my mobile device, the app shows the Trying to connect popup but the browser seems unaffected by this issue.

I was going for an attempt on the todos app to get a feel of meteor 1.3, but it seems i have to postpone it until the app works straight out of the repository on my device.

these are my console errors ( i hope someone can help make it work):
`I20160426-02:19:32.403(2)? E/chromium(15632): [ERROR:xwalk_autofill_client.cc(172)] Not implemented reached in virtual void xwalk::XWalkAutofillClient::OnFirstUserGestureObserved()
I20160426-02:19:39.063(2)? E/chromium(15632): [ERROR:gin_java_bridge_message_filter.cc(106)] WebView: Unknown frame routing id: 1
I20160426-02:19:39.493(2)? I/chromium(16435): [INFO:library_loader_hooks.cc(143)] Chromium logging enabled: level = 0, default verbosity = 0
I20160426-02:19:40.542(2) (android:file:///android_asset/www/meteor_cordova_loader.js:12) METEOR CORDOVA DEBUG (meteor_cordova_loader.js) Loading from url: http://meteor.local
I20160426-02:19:41.629(2) (android:http://meteor.local/packages/autoupdate.js:338) METEOR CORDOVA DEBUG (autoupdate_cordova.js) cordova.file.dataDirectory is null, retrying in 20ms`

**Comment by martijnwalraven** [comment]
@timmakken: Are you sure you are running Meteor 1.3? Those logs refer to `meteor.local`, which is not used in the current version of the Meteor Cordova plugin.

**Comment by timmakken** [comment]
Did git clone to this repo a few minutes ago , and the release file states `METEOR@1.3
` so i think so yes :+1:

**Comment by martijnwalraven** [comment]
Well, those logs are definitely not from Meteor 1.3. So either they are from another app running on the same device (`logcat` logs aren't filtered per app) or something is wrong with your install.

Have you tried remote debugging the app to see if you get any errors in the browser console?

**Comment by timmakken** [comment]
Well there was another app running on another port but it was not running on my device so im very confused about how a git clone -> meteor npm install -> meteor run android-device can result in me having an old version of meteor while i pulled it from the master branch today.

Can you help me find out by explaining how to do remote debugging to the app? ( mayb a link )

**Comment by martijnwalraven** [comment]
Yes, I'm also confused! Could there be another app installed on your device that is still running in the background?

Here is a link to the [section of the mobile guide on logging and remote debugging](http://guide.meteor.com/mobile.html#logging-and-remote-debugging).

**Comment by timmakken** [comment]
Found the culprit!  Turns out i had an old version of the todos app installed and did not update properly.
After removal of the old app and reinstall it worked.

I remember something sashko said about having to push a new version to the app store when migrating to 1.3 because of the new cordova implementation which gave me the idea.

**Comment by dagatsoin** [comment]
Just add my 2 cts on the need to remove any previous cordova-build before rebuild a new one. Otherwise I get stuck on random steps (sometimes "Determining Active Plugins", other times "Compiling Ecmascript"...)

**Comment by dgroch** [comment]
@martijnwalraven @tmeasday I just hit this issue after upgrading to `release/METEOR@1.4.2-beta.4` (needed to get builds passing for iOS 10)

**Comment by hwillson** [comment]
I've confirmed the `shell.js: internal errorject` error is being caused by the `shelljs` version 0.6.0 dependency specified by the `eslint` package. Until this dependency is updated by the eslint project themselves, the easiest (but not greatest) workaround is to add a more recent shelljs `devDependency` to the todos app. I've tested this approach with `shelljs` version 0.7.4, and it works properly. I'll submit a PR to address this.
